Tests/ApolloCacheDependentTests/LoadQueryFromStoreTests.swift

```diff
@@ -298,26 +298,15 @@ class LoadQueryFromStoreTests: XCTestCase, CacheTesting {
             XCTFail("Incorrect error type for primary error: \(error)")
             return
           }
-          
-          switch graphQLError.underlying {
-          case is JSONDecodingError:
-            if (cache is InMemoryNormalizedCache) {
-              // This is expected for in-memory caching
-              break
-            } else {
-              XCTFail("Incorrect error type for underlying with in-memory cache: \(graphQLError.underlying)")
-            }
-          #if canImport(ApolloSQLite)
-          case is SQLiteNormalizedCacheError:
-            if (cache is SQLiteNormalizedCache) {
-              // This is expected for SQLite caching
-              break
-            } else {
-              XCTFail("Incorrect error type for underlying with SQLite cache: \(graphQLError.underlying)")
-            }
-          #endif
-          default:
-            XCTFail("Incorrect error type for underlying: \(graphQLError.underlying)")
+          guard let decodingError = graphQLError.underlying as? JSONDecodingError else {
+            XCTFail("Invalid error type.")
+            return
+          }
+          switch decodingError {
+          case .couldNotConvert(value: _, to: _):
+            break
+          case .missingValue, .nullValue, .wrongType:
+            XCTFail("Invalid error type.")
```

Comment:
I think you can just do this and have this be a lot simpler: 

```swift
switch graphQLError.underlying {
case JSONDecodingError.couldNotConvert(value: _, to: _)?:
  break
default: 
   XCTFail("Invalid error type")
}

Reply:
@designatednerd - The above change was made, but it looks like CircleCI is complaining about `ci/circleci: Apollo iOS 12.4` build.  I ran the tests locally and didn't see any issues.  I'm unable to trigger the build process again to see if it might pass and it was a glitch.  My previous code passed all the tests on all the CircleCI checks.

Reply:
Yeah there's one test that gets flaky a decent amount - I'll kick the build